### DISCUSSION TITLE
Reducing the check time?

### DISCUSSION THREAD

**Discussion by 3v1n0** [upptime/upptime#648] [Ideas]
While github schedules can be triggered every 5 minutes, why not making each instance to stay alive for the whole period to perform the checks more frequently?

**Comment by AnandChowdhary** [comment]
This is actually a pretty neat idea! Each job can last for up to six hours, so we can run the workflow for six hours, and after maybe 5 hours and 50 minutes we can trigger the same workflow again using the currently running workflow to keep it alive forever.

**Reply by 3v1n0** [reply]
Yeah, maybe it's possible to enable this if users have set frequent checks, not to use runners time, but in terms of energy usage it should be even better as there's no computation involved while waiting and there's less ops to do when starting/stop a new instance and cloning data.

**Comment by Andre601** [comment]
I personally dislike this idea... It abuses the Actions system more than it already is because... even tho this is a neat setup, the Actions are (at least in my eyes) bordering to abuse of what GitHub provides.

**Reply by Andre601** [reply]
Also, to add to this: From what I understand and know is GitHub not having an infinite amount of runners. Action workflows are "picked up" by any runner that happens to be available for use, so if every repo using uptime would use a runner for up to 5+ hours could this easily fill up the available runners GitHub has available and may even cause technical issues in the long run.

No Action should ever run that long. It is surprising to me that GitHub *does* allow to use a runner for up to 6 hours.
It is simply abuse of the system in this case. Action should start, do its thing and then finish... Keeping it open artificially is not a good option at all.

**Reply by AnandChowdhary** [reply]
Now that I think about it, I agree. Let me ask some folks over at GitHub and see what they think!
